Core/ChatCommands: Check whether a passed numeric enum value is valid

Carbenium opened the pull request:
**Changes proposed:**

-  Core/ChatCommands: Check whether a passed numeric enum value is valid

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

Carbenium committed:
Core: Make EncounterState enum smart

Carbenium committed:
Common/SmartEnum: Add helper to check if enum value is valid

Carbenium committed:
Core/ChatCommands: Check whether a passed numeric enum value is valid

Carbenium committed:
Contrib/EnumUtils: Use nested namespaces in generated file

Carbenium committed:
Common/EnumUtils: Add EnumUtils::ToIndex(Enum)

Carbenium commented on src/common/Utilities/SmartEnum.h:
```diff
@@ -23,7 +23,9 @@
 struct EnumText
 {
-    EnumText(char const* c, char const* t, char const* d) : Constant(c), Title(t), Description(d) {}
+    EnumText(uint32 i, char const* c, char const* t, char const* d) : Index(i), Constant(c), Title(t), Description(d) { }
```
I'm not quite happy with `EnumText` and `ToString` anymore if we do it like this.

@Treeston suggestions?

Treeston replied:
no need to add it to enumtext

literally just make it the opposite of what FromIndex already is

Carbenium committed:
Scripts/Commands: No need to check for valid enum value

The argument parser already checks for validity and the
handler does not get executed it it's invalid.
